src/expr/type_node.cpp

```diff
@@ -558,6 +561,15 @@ bool TypeNode::isSygusDatatype() const
   return false;
 }
 
+bool TypeNode::isAbstract() const { return getKind() == kind::ABSTRACT_TYPE; }
+
+Kind TypeNode::getAbstractKind() const
```

Comment:
Maybe this should be renamed, the name is a bit misleading. This returns the kind of the sort arameter, no?

Reply:
It is called on abstract sorts, and returns the kind that they abstract.  e.g. calling this on the type `ABSTRACT_TYPE{BITVECTOR_TYPE}` would return `BITVECTOR_TYPE`.

Maybe, `getAbstractedType`?

Reply:
Yes, sounds good.